shared/domains/chelonia/chelonia.js

```diff
@@ -852,64 +893,67 @@ export default (sbp('sbp/selectors/register', {
     if (!rootState[contractID]) this.config.reactiveSet(rootState, contractID, state)
     const originatingState = originatingContract.state(originatingContractID)
     const { HEAD: previousHEAD, height: previousHeight } = await sbp('chelonia/private/out/latestHEADinfo', contractID)
-    const innerSigningKey = this.transientSecretKeys[innerSigningKeyId]
-    const signedInnerData = [originatingContractID, encryptionKeyId, outerKeyId, GIMessage.OP_KEY_REQUEST, contractID, previousHEAD]
-    signedInnerData.forEach(x => { if (x.includes('|')) { throw Error(`contains '|': ${x}`) } })
+
+    const keyRequestReplyKey = keygen(EDWARDS25519SHA512BATCH)
+    const keyRequestReplyKeyId = keyId(keyRequestReplyKey)
+    const keyRequestReplyKeyP = serializeKey(keyRequestReplyKey, false)
+    const keyRequestReplyKeyS = serializeKey(keyRequestReplyKey, true)
+
+    const signingKeyId = findSuitableSecretKeyId(originatingState, [GIMessage.OP_KEY_ADD], ['sig'])
+    if (!signingKeyId) {
+      throw ChelErrorUnexpected(`Unable to send key request. Originating contract is missing a key with OP_KEY_ADD permission. contractID=${contractID} originatingContractID=${originatingContractID}`)
+    }
+    await sbp('chelonia/out/keyAdd', {
+      contractID: originatingContractID,
+      contractName: originatingContractName,
+      data: [{
+        id: keyRequestReplyKeyId,
+        name: 'krrk-' + keyRequestReplyKeyId,
```

Comment:
Either needs to be done now or later, all internal key names (and reserved keys) should be prefixed with `#`